RoomLocator/RoomLocator.Data/Services/ScadadataService.cs

```diff
@@ -0,0 +1,128 @@
+            var lights = scadadataViewModels.Where(item => item.Type == "Light").ToList(); 
+            var sounds = scadadataViewModels.Where(item => item.Type == "Sound").ToList(); 
+            var availableSeats = scadadataViewModels.Where(item => item.Type == "available seats").ToList(); 
+            var maxAvailableSeats = scadadataViewModels.Where(item => item.Type == "max available seats").ToList();
+            var list = new List<ScadadataScoresModel>
+            {
+                new ScadadataScoresModel("Temperature",
+                    1 - FindAverageDeviation(temperatures, TemperatureOptimal, TemperatureMaxVariance)),
+                new ScadadataScoresModel("Sound",
+                    1-FindAverageDeviation(sounds, SoundOptimal, SoundMaxVariance)),
+                new ScadadataScoresModel("Light",
+                    1-FindAverageDeviation(lights, LightOptimal, LightMaxVariance)),
+                new ScadadataScoresModel("Seats Available",
+                    1-FindAverageDeviation(availableSeats, maxAvailableSeats[0].Value, 
+                        maxAvailableSeats[0].Value-40))
+            };
+            return list;
+
+
+        }
+        public async Task<double> GetWeightedScore()
+        {
+            var sd = await GetSensorData();
+            var scadadataViewModels = sd.ToList();
+            var temperatures = scadadataViewModels.Where(item => item.Type == "Temperature").ToList();
+            var lights = scadadataViewModels.Where(item => item.Type == "Light").ToList(); 
+            var sounds = scadadataViewModels.Where(item => item.Type == "Sound").ToList(); 
+            var availableSeats = scadadataViewModels.Where(item => item.Type == "available seats").ToList(); 
+            var maxAvailableSeats = scadadataViewModels.Where(item => item.Type == "max available seats").ToList();
+            
+            return (1-FindAverageDeviation(temperatures, TemperatureOptimal, TemperatureMaxVariance))*TemperatureWeight
+                   +(1-FindAverageDeviation(sounds, SoundOptimal, SoundMaxVariance))*SoundWeight
+                   +(1-FindAverageDeviation(lights, LightOptimal, LightMaxVariance))*LightWeight
+                   +(1-FindAverageDeviation(availableSeats, maxAvailableSeats[0].Value, 
+                         maxAvailableSeats[0].Value))*SeatsWeight;
+        }
+
+        private static double FindAverageDeviation(IEnumerable<ScadadataViewModel> list, double optimal, double maxVariance)
+        {
+            var deviations = list.Select(item => GetDeviation(item.Value, optimal, maxVariance)).ToList();
+            return deviations.Average();
+        }
+        
+        private static double GetDeviation(double measurement, double optimal, double maxVariance)
+        {
+            var deviation = (Math.Abs(optimal - measurement)) / maxVariance ;
+            return deviation > 1.0 ? 1.0 : deviation;
+        }
+
+
+        
+    }
+}
+
+
+
+
+     
+
+
+
+
+
+
+
+      
+
+    
```

Comment:
Normalt kommenterer jeg ikke på styling, men når du allerede er i gang med at ædnre klassen, så fjern gerne alt det her ekstra mellemrum også.

Hvis det er for at undgå at teksten står i bunden af din editor, så findes der extensions til de fleste editors/ide'er der lader en scrolle længere end filens længde. Jeg kan nemlig heller ikke selv tage når teksten står i bunden af skærmen